pkg/store/cas/cas.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cas
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	ariesstorage "github.com/hyperledger/aries-framework-go/spi/storage"
+	"github.com/ipfs/go-cid"
+	mh "github.com/multiformats/go-multihash"
+)
+
+// CAS represents a content-addressable storage provider.
+type CAS struct {
+	cas ariesstorage.Store
+}
+
+// New returns a new CAS that uses the passed in provider as a backing store.
+func New(provider ariesstorage.Provider) (*CAS, error) {
+	cas, err := provider.OpenStore("cas_store")
+	if err != nil {
+		return nil, fmt.Errorf("failed to open store in underlying storage provider: %w", err)
+	}
+
+	return &CAS{cas: cas}, nil
+}
+
+// Write writes the given content to the underlying storage provider.
+// Returns the SHA256 hash in base64url encoding which represents the address of the content.
+func (p *CAS) Write(content []byte) (string, error) {
+	// TODO #318 figure out why the CIDs produced here differ from the ones that IPFS generates.
+	prefix := cid.Prefix{
+		Version:  0,
+		MhType:   mh.SHA2_256,
+		MhLength: -1, // default length
+	}
+
+	contentID, err := prefix.Sum(content)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate CID: %w", err)
+	}
+
+	err = p.cas.Put(contentID.String(), content)
+	if err != nil {
+		return "", fmt.Errorf("failed to put content into underlying storage provider: %w", err)
+	}
+
+	return base64.URLEncoding.EncodeToString(contentID.Bytes()), nil
```

Comment:
you should just return contentID.String() - no encoding

Reply:
Done!